crates/types/src/response.rs

```diff
@@ -33,6 +40,13 @@ impl Response {
         self
     }
 
+    pub fn may_add_submessage(mut self, maybe_submsg: Option<SubMessage>) -> Self {
```

Comment:
True but I feel `may_add_message` is clearer in its intention. `add_messages` is intended for adding more than one messages in one go.

Reply:
Yep, it was just a clarification. Let's say that having a dedicated method for options avoids turning an option into an iterator, it could save some gas